enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,309 @@
+
+## Proposal
+
+Log forwarding will provide a declarative way to specify the destination for specific types of logs using a 'pipeline'.  A 'pipeline' defines simple routing for one source to one or more destinations. The source of logs are opinionated and well defined by cluster logging. The initial source types are as follows:
+
+* `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
+* `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
+
+Future types may be:
+* `events.kubernetes` - Kubernetes events
+* `audit.container-engine` - Container engine audit events
+
+There are no assumptions regarding whether or not an endpoint is deployed on or off cluster.  Endpoints off-cluster may require adminstrators to perform additional actions in order for logs to be forwarded (e.g. secret creation, opening port) 
+Following is the list of supported endpoint types for this proposal:
+
+* Elasticsearch
+* Fluent secure-forward
+
+### User Stories
+
+#### As an OKD admin, I want to collect logs and send them to my own aggregation solution, for example Splunk
+
+This is a typical example of organizations that desires to re-use their existing enterprise log solution.  We will succeeded if we are able to write logs to their logging service.
+
+#### As an OKD admin, I want to aggregate application logs on-cluster and infra logs off-cluster
+
+This is an example of an OKD cluster hosting solution where several organizations are each provided with a dedicated cluster.  The organization requires access to application container logs but the host requires access to the infra structure logs.
+
+### Implementation Details
+
+#### Assumptions
+
+* A pipeline shall support at least one target endpoint. 
+* A pipeline may support multiple target endpoints.
+* Endpoints that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
+* Secrets shall have keys of: cert,key,cacert which point to the respective certificates for which they represent
+
+#### Security
+* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: cert, key, cacert
+* Endpoints defined without a secret will use unsecure communication
+
+#### Cluster Logging With Logstore, Collector, and LogForwarding
+* Undeclared source messages are routed to logstore
```

Comment:
Is there really much of a difference (implementation-wise) between this scenario and the following?
Could we just default that no declared pipeline means we default all source to the on cluster log store, otherwise if a pipeline is defined then to send logs to the created logstore users would need to be explicit as well and specify a constant that the operator would know means to route to the logstore?

Reply:
what is an "undeclared source message"?  what is "logstore"?

Reply:
I think this means that if you have deployed the full OpenShift logging stack with the Logstore (i.e. Elasticsearch), and you have specified a source and target for `logs.app`, but did not specify anything for `logs.infra`, then logs from `logs.infra` will automatically be stored in the OpenShift logging stack Logstore with no further configuration required.

Reply:
ok. as long as we have good docs :)

"defaults to internal logstore if you have one, vanishes to devnull otherwise"